src/Core/Content/Product/SalesChannel/Listing/Processor/SortingListingProcessor.php

```diff
@@ -95,30 +94,22 @@ private function getAvailableSortings(Request $request, Context $context): Produ
         $criteria->setTitle('product-listing::load-sortings');
         /** @var string[] $availableSortings */
         $availableSortings = $request->get('availableSortings');
-        $availableSortingsById = [];
 
         if ($availableSortings) {
-            arsort($availableSortings, \SORT_DESC | \SORT_NUMERIC);
-            $availableSortingsFilter = array_keys($availableSortings);
+            $sortings = new ProductSortingCollection();
+            /** @var ProductSortingEntity $availableSorting */
+            foreach ($availableSortings as $availableSorting) {
```

Comment:
you should put the entities under another request parameter. you are now even need to overwrite the var annotation from above, as only a list of IDs was expected